### PR TITLE
Make it work with Github CI

### PR DESCRIPTION
**Pull request by BenoitAnastay**
I adapted the code to make it working with the currently used Github CI

I'll comment the code to explains what I changed and why

As you can see the build passe on both Travis and Github

### CODE REVIEW

File: .github/workflows/githubci.yml [hunk 1]
```diff
@@ -9,19 +9,20 @@ jobs:
     
     steps:
     - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x' # Version range or exact version of a Python version to use, using semvers version range syntax.
+        architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
```
Comment (BenoitAnastay) [inline .github/workflows/githubci.yml]: Force python version 3 on github

File: .github/workflows/githubci.yml [hunk 2]
```diff
@@ -9,19 +9,20 @@ jobs:
...
-        build_platform esp8266
-        build_platform m4
-        build_platform nrf52840
+        ./build.sh build_platform esp32
+        ./build.sh build_platform uno
+        ./build.sh build_platform leonardo
+        ./build.sh build_platform zero
+        ./build.sh build_platform esp8266
+        ./build.sh build_platform m4
+        ./build.sh build_platform nrf52840
```
Comment (BenoitAnastay) [inline .github/workflows/githubci.yml]: Made separate file for build because github was unable tu run it from the sourced one

File: .travis.yml [hunk 3]
```diff
@@ -14,19 +14,18 @@ addons:
     packages:
       - python3
       - python3-pip
-      - python3-setuptools
```
Comment (BenoitAnastay) [inline .travis.yml]: Moved it in install.sh because it was missing with this way

File: .travis.yml [hunk 4]
```diff
@@ -14,19 +14,18 @@ addons:
...
-  - build_platform zero
-  - build_platform esp8266
-  - build_platform m4
+  - $TRAVIS_BUILD_DIR/build.sh build_platform nrf52840
+  - $TRAVIS_BUILD_DIR/build.sh build_platform esp32
+  - $TRAVIS_BUILD_DIR/build.sh build_platform uno
+  - $TRAVIS_BUILD_DIR/build.sh build_platform leonardo
+  - $TRAVIS_BUILD_DIR/build.sh build_platform zero
+  - $TRAVIS_BUILD_DIR/build.sh build_platform esp8266
+  - $TRAVIS_BUILD_DIR/build.sh build_platform m4
```
Comment (BenoitAnastay) [inline .travis.yml]: Same as before separate build even if it was working on travis it's better to have all on the same place

**Comment by ladyada** [comment]
nice! do you mind removing the whitespace-only changes on the other files?

File: doxy_gen_and_deploy.sh [hunk 5]
```diff
@@ -1,162 +1,162 @@
-#!/bin/bash
-################################################################################
```
Comment (BenoitAnastay) [inline doxy_gen_and_deploy.sh]: This file didn't changed I don't know why Github display that

File: install.sh [hunk 6]
```diff
@@ -79,6 +79,7 @@ ln -s $TRAVIS_BUILD_DIR $HOME/arduino_ide/libraries/Adafruit_Test_Library
 
 # add the arduino CLI to our PATH
 export PATH="$HOME/arduino_ide:$PATH"
+export PATH="$HOME/.local/bin/:$PATH"
```
Comment (BenoitAnastay) [inline install.sh]: I tried to add that so github would exec nrfutils but didn't worked, I had to sudo install them

File: install.sh [hunk 7]
```diff
@@ -135,7 +136,12 @@ fi
 
 if [[ $INSTALL_NRF52 == 1 ]]; then
   echo -n "ADAFRUIT NRF5X: "
+  pip3 install --user setuptools
   pip3 install --user adafruit-nrfutil
+  pip3 install --user pyserial
```
Comment (BenoitAnastay) [inline install.sh]: pyserial was sometimes missing

File: install.sh [hunk 8]
```diff
@@ -135,7 +136,12 @@ fi
 
 if [[ $INSTALL_NRF52 == 1 ]]; then
   echo -n "ADAFRUIT NRF5X: "
+  pip3 install --user setuptools
```
Comment (BenoitAnastay) [inline install.sh]: setuptools here because by apt it was missing on github

File: install.sh [hunk 9]
```diff
@@ -135,7 +136,12 @@ fi
...
   pip3 install --user adafruit-nrfutil
+  pip3 install --user pyserial
+  sudo pip3 install setuptools
+  sudo pip3 install adafruit-nrfutil
+  sudo pip3 install pyserial
```
Comment (BenoitAnastay) [inline install.sh]: sudo install was the only way to make it work with github

File: library_check.sh [hunk 10]
```diff
@@ -1,3 +1,3 @@
-#!/bin/bash
-
-# empty for now, add QA checkin' later!
+#!/bin/bash
```
Comment (ladyada) [inline library_check.sh]: please revert this file (whitespace only changes)

File: doxy_index.html [hunk 11]
```diff
@@ -1,11 +1,11 @@
...
-<html lang="en-US">
-    <head>
-       <meta charset="UTF-8">
-        <meta http-equiv="refresh" content="1;url=html/index.html">
-        <title>Page Redirection</title>
-    </head>
-    <body>
-        If you are not redirected automatically, follow the <a href="html/index.html">link to the documentation</a>
-    </body>
+<!DOCTYPE HTML>
```
Comment (ladyada) [inline doxy_index.html]: please revert this file (whitespace only changes)

**Comment by BenoitAnastay** [comment]
@ladyada I made no changes on the others files I don't know what happenned i'll try to revert those

File: doxy_gen_and_deploy.sh [hunk 12]
```diff
@@ -1,162 +1,162 @@
...
-    # that might otherwise be exposed.
-    echo 'Git pushing'
-    git push --force "https://${GH_REPO_TOKEN}@github.com/${TRAVIS_REPO_SLUG}.git" > /dev/null 2>&1
-else
-    echo '' >&2
-    echo 'Warning: No documentation (html) files have been found!' >&2
-    echo 'Warning: Not going to push the documentation to GitHub!' >&2
-    exit 1
-fi
+#!/bin/bash
```
Comment (ladyada) [inline doxy_gen_and_deploy.sh]: please revert this file (whitespace only changes)

File: Doxyfile.default [hunk 13]
```diff
@@ -1,2458 +1,2458 @@
...
-# Note that a value of 0 will completely suppress the enum values from appearing
-# in the overview section.
-# Minimum value: 0, maximum value: 20, default value: 4.
-# This tag requires that the tag GENERATE_HTML is set to YES.
-
-ENUM_VALUES_PER_LINE   = 4
-
-# If the treeview is enabled (see GENERATE_TREEVIEW) then this tag can be used
-# to set the initial width (in pixels) of the frame in which the tree is shown.
-# Minimum value: 0, maximum valu
```
Comment (ladyada) [inline Doxyfile.default]: please revert this file (whitespace only changes)

**Comment by ladyada** [comment]
thanks, i marked which ones they are

**Comment by ladyada** [comment]
thank you! last question, there's build.sh - can we not use install.sh?

**Comment by BenoitAnastay** [comment]
@ladyada Github do not seem to be able to call a sourced function and after trying to make a messy condition into install.sh i thought it was a better way to separate them but if you know a clean way to merge them, still calling build from argument an not repeating install, I can apply it

I was thinking of something like this 
```
If [No arguments]
Then
 //Install
Else
//Declare functions
//call function argv
If
```

But I didn't manage to make it work, I do not tried a lot because I thought it was messy
```

**Comment by BenoitAnastay** [comment]
This way it throw Permission denied on both Github and Travis

**Comment by BenoitAnastay** [comment]
Ok maybe a chmod issue I'll try again

**Comment by BenoitAnastay** [comment]
`mode change 100644 => 100755 install.sh`

Now this should work, it does on my local machine

**Comment by BenoitAnastay** [comment]
@ladyada 
So it work on both Github and travis with only install.sh file 

[Build Status](https://travis-ci.org/BenoitAnastay/travis-ci-arduino)
[Build Status](https://github.com/BenoitAnastay/travis-ci-arduino/actions)

**Comment by ladyada** [comment]
ok we'll try it! thank you so much :)

**Comment by BenoitAnastay** [comment]
It worked, the commit have the green mark, nice

**Pull request merged**